pkg/registry/core/pod/storage/eviction.go

```diff
@@ -291,7 +292,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	}
 
 	// Try the delete
-	err = addConditionAndDeletePod(r, ctx, eviction.Name, rest.ValidateAllObjectFunc, deleteOptions)
+	err = addConditionAndDeletePod(r, ctx, eviction.Name, pod, deleteOptions)
 	if err != nil {
 		if errors.IsConflict(err) && updateDeletionOptions &&
```

Comment:
if `updateDeletionOptions == false`, and we hit a conflict adding the condition, do we retry or return an error to the user? if we return an error, is it an error that prompts a retry as commented in this block or not?

Reply:
After implementing the https://github.com/kubernetes/kubernetes/pull/116554/files/4ca6333cc16e3995c3b3a3a6b158ad2d9f53d845#diff-6105105bc4d40e3d45b9cc38165516be0205f0329929421fd7df65df7abaeb2c to remove the conflict race, it seems to me we can keep the old logic here. That is, retry only if we made an eviction decision based on stale information (readiness of the pod), that could have changed  in the meantime.